app/controllers/smart_proxies_controller.rb

```diff
@@ -67,11 +64,25 @@ def destroy
 
   private
 
+  def requested_data(data_name)
+    data = @smart_proxy.public_send(data_name)
+    render :json => {:success => true, :message => data}
+  rescue Foreman::Exception => exception
```

Comment:
i think we need to catch more exceptions here (at somewhere else) I just had Errno::ENOENT  exception (No such file or directory @ rb_sysopen when opening ssl certificate)